code/game/objects/items/weapons/mop.dm

```diff
@@ -28,18 +28,20 @@
 	GLOB.janitorial_equipment -= src
 	return ..()
 
-/obj/item/mop/proc/wet_mop(obj/o, mob/user)
-	if(o.reagents.total_volume < 1)
-		to_chat(user, "[o] is out of water!</span>")
-		if(istype(o, /obj/structure/mopbucket))
-			mopbucket_insert(user, o)
-		if(istype(o, /obj/structure/janitorialcart))
-			janicart_insert(user, o)
+/obj/item/mop/proc/wet_mop(obj/O, mob/user)
+	if(O.reagents.total_volume < 1)
+		to_chat(user, "[O] is out of water!</span>")
```

Comment:
```suggestion
		to_chat(user, "[O] is empty!</span>")
```

Object doesn't necessarily need to be filled with water, so a more generic message might be appropriate.